plugins/quota/validate.go

```diff
@@ -3,38 +3,65 @@ package quota
 import (
 	"fmt"
 
+	daemon "github.com/gluster/glusterd2/glusterd2/daemon"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
 )
 
 const (
-	quotaEnabledKey = "features.quota"
+	quotaEnabledKey = "quota.enable"
 )
 
 func validateOptions(v *volume.Volinfo, key string, value string) error {
 
-	if isQuotaEnabled(v) == false {
-		err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
-		return err
+	// Check if volume is started
+	if v.State != volume.VolStarted {
+		return errors.ErrVolNotStarted
 	}
 
 	switch key {
-	case "deem-statfs":
+	case "enable":
+		switch value {
+		case "on":
+			ret, err := isQuotadRunning()
+			if err == nil && ret {
+				// As quotad for various process share a
+				// single pid, it is necessary to check the
+				// options too.
+				if isQuotaEnabled(v) {
+					return errors.ErrProcessAlreadyRunning
+				}
+
+			}
+		case "off":
+			ret, err := isQuotadRunning()
+			if !ret {
+				return err
+			}
+		}
 		return nil
+	case "deem-statfs":
+		fallthrough
 	case "hard-timeout":
-		return nil
+		fallthrough
 	case "soft-timeout":
-		return nil
+		fallthrough
 	case "alert-time":
-		return nil
+		fallthrough
 	case "default-soft-limit":
-		return nil
+		if isQuotaEnabled(v) == false {
```

Comment:
for all the fallthough this condition is checked. is it expected?

Reply:
yes